stdlib/camlinternalFormat.ml

```diff
@@ -1500,8 +1505,8 @@ fun k o acc fmt -> match fmt with
     make_int_padding_precision k o acc rest pad prec convert_int64 iconv
   | Float (fconv, pad, prec, rest) ->
     make_float_padding_precision k o acc rest pad prec fconv
-  | Bool rest ->
-    fun b -> make_printf k o (Acc_data_string (acc, string_of_bool b)) rest
+  | Bool (pad, rest) ->
+    make_string_padding k o acc rest pad string_of_bool
```

Comment:
I would be a bit nicer to rename this `make_string_padding` function into `make_padding`, given that it is not string-specific anymore. But the current version is correct (it so happens that the string function had just the right amount of genericity to support booleans as well).

Reply:
Renamed.